athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/SchemaBuilder.java

```diff
@@ -279,6 +280,22 @@ public SchemaBuilder addMetadata(String key, String value)
         return this;
     }
 
+    /**
+     * Adds the provided metadata to the Schema, unless the provided key is one of the values in excludedKeys
+     *
+     * @param key The key of the metadata to add.
+     * @param value The value of the metadata to add.
+     * @param excludeKeys set of strings that should be excluded from being added to the metadata map
+     * @return This SchemaBuilder itself.
+     */
+    public SchemaBuilder addMetadata(String key, String value, Set<String> excludeKeys)
```

Comment:
This method strikes me as fairly specific / odd. What would be an example of an excludedKey? and why would we be attempting to addMetadata for the same key twice?

Reply:
Ok, i see where it is used further down. My advice is to prevent SchemaBuilder from allowing the same key to be set multiple times. You may need to add a set or something else to track keys since the Arrow object might not be helpful for this. Then, in the DDB connector or Glue handler, set the keys you want to treat as reserved first. Then have the addMetadata function throw 'DuplicateMetadtaException' if a key gets set more than once.

Reply:
its not that I want to add the same key twice (which is already not possible as metadata is ImmutableMap). When we read in table parameters, we currently copy everything field to the schema metadata. I want to change this so that we exclude some keys from being copied over. I then later add a "normalized value" for the skipped key to metadata.

Reply:
I think my view is still the same, I think this method makes using SchemaBuilder more confusing and would recommend we handle this outside the builder. Would that be possible or painful? I'm not sure I've seen this usecase in any other connector yet which is also why I'm apprehensive to change this in the SDK.